staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -390,6 +390,16 @@ func (r *crdHandler) getServingInfoFor(crd *apiextensions.CustomResourceDefiniti
 	return ret, nil
 }
 
+// crdObjectConverter is a converter that supports field selectors for CRDs.
+type crdObjectConverter struct {
+	unstructured.UnstructuredObjectConverter
+}
+
+func (crdObjectConverter) ConvertFieldLabel(version, kind, label, value string) (string, string, error) {
+	// Just return the passed-in label and value, as CRDs only support a single version currently.
```

Comment:
For other types, we return an error at this point if the field label isn't in a whitelist:

https://github.com/kubernetes/kubernetes/blob/2bf7a20a542ac54ccb6c284aa0b01559bf02a43b/pkg/api/v1/conversion.go#L177

If we don't error out, it just comes back as an empty list, which is misleading.

Whitelisting metadata.name and metadata.namespace would also mean we can remove the caveat about this only working because CRD currently supports only one version -- at least until ObjectMeta v2.

Reply:
@enisoc are you saying this should just return the result of k8s.io/apimachinery/pkg/runtime/conversion.go#DefaultMetaV1FieldSelectorConversion()?

Reply:
Hm the behavior there seems reasonable, although it's slightly different than what other resources do. Other cluster-scoped resources seem to error out on `metadata.namespace`:

https://github.com/kubernetes/kubernetes/blob/31697a27f9e7ff5f7f80f61955106d2f33e9165b/pkg/api/v1/conversion.go#L184-L195

Reply:
We know if it's cluster-scoped or not, so I'll just write a custom function

Reply:
@enisoc just pushed an update - ptal!